src/main/java/org/terasology/logic/health/HealthAuthoritySystem.java

```diff
@@ -140,16 +165,199 @@ private int calculateTotal(int base, TFloatList multipliers, TIntList modifiers)
+            }
+        }
+    }
+
+    /**
+     * Handles DoDamageEvent to inflict damage to entity with HealthComponent.
+     *
+     * @param event DoDamageEvent causing the damage on the entity.
+     * @param entity The entity which is damaged.
+     */
+    @ReceiveEvent
+    public void onDamage(DoDamageEvent event, EntityRef entity) {
+        checkDamage(entity, event.getAmount(), event.getDamageType(), event.getInstigator(), event.getDirectCause());
+    }
+
+    private void checkDamage(EntityRef entity, int amount, Prefab damageType, EntityRef instigator, EntityRef directCause) {
+        BeforeDamagedEvent beforeDamage = entity.send(new BeforeDamagedEvent(amount, damageType, instigator, directCause));
+        if (!beforeDamage.isConsumed()) {
+            int damageAmount = TeraMath.floorToInt(beforeDamage.getResultValue());
+            if (damageAmount > 0) {
+                doDamage(entity, damageAmount, damageType, instigator, directCause);
+            } else {
+                doHeal(entity, -damageAmount, instigator);
+            }
+        }
+    }
+
+    /**
+     * Handles damage sound on inflicting damage to entity.
+     *
+     * @param event OnDamagedEvent triggered when entity is damaged.
+     * @param entity Entity which is damaged.
+     * @param characterSounds Component having sound settings.
+     */
+    @ReceiveEvent
+    public void onDamaged(OnDamagedEvent event, EntityRef entity, CharacterSoundComponent characterSounds) {
+        if (characterSounds.lastSoundTime + CharacterSoundSystem.MIN_TIME < time.getGameTimeInMs()) {
+
+            // play the sound of damage hitting the character for everyone
+            DamageSoundComponent damageSounds = event.getType().getComponent(DamageSoundComponent.class);
+            if (damageSounds != null && !damageSounds.sounds.isEmpty()) {
+                StaticSound sound = random.nextItem(damageSounds.sounds);
+                if (sound != null) {
+                    entity.send(new PlaySoundEvent(sound, 1f));
+                }
+            }
+
+            // play the sound of a client's character being damaged to the client
+            if (!characterSounds.damageSounds.isEmpty()) {
+                StaticSound sound = random.nextItem(characterSounds.damageSounds);
+                if (sound != null) {
+                    entity.send(new PlaySoundForOwnerEvent(sound, characterSounds.damageVolume));
+                }
+            }
+
+            characterSounds.lastSoundTime = time.getGameTimeInMs();
+            entity.saveComponent(characterSounds);
+
+        }
+    }
+
+    /**
+     * Causes damage to entity when fallingDamageSpeedThreshold is breached.
+     *
+     * @param event VerticalCollisionEvent sent when falling speed threshold is crossed.
+     * @param entity The entity which is damaged due to falling.
+     */
+    @ReceiveEvent(components = {HealthComponent.class})
```

Comment:
Since the `HealthComponent` is used inside this event handler you should be able to move it from the `@ReceiveEvent` filter into a method parameter. That way you don't need to separately retrieve it 👍